# Mobile genetic elements (MGEs) in gut metagenomes

In the beginning the mobile genetic elements were associated with pathogenic strains; however, this has changed with time [Dobrindt et al.2004](https://pubmed.ncbi.nlm.nih.gov/15100694/). For example, an immense diversity of phages has been described in the ocean, suggesting that these mobile elements might play a relevant role in microbial communities working as seed banks for adaptive genes, which any community member can potentially acquire ([Fancello et al.2011](https://pubmed.ncbi.nlm.nih.gov/21816767/); [Modi et al.2013](https://pubmed.ncbi.nlm.nih.gov/23748443/); [Subirats et al.2016](https://pubmed.ncbi.nlm.nih.gov/27312355/)).

I started my search by looking for these three types of mobile elements.

- [Insertion sequences](https://github.com/mirnavazquez/gut_microbiota/blob/main/Rmarkdown/2022_04_04-ISEScan-Searching_IS_elements.Rmd)
- [Integrons](https://github.com/mirnavazquez/gut_microbiota/blob/main/Rmarkdown/2022_04_04-IntegronFinder-Searching_integrons.Rmd)
- [Phages](https://github.com/mirnavazquez/gut_microbiota/blob/main/Rmarkdown/2022_04_04-PHASTER-Looking_for_phages.Rmd)

The scripts are in a directory called [R](https://github.com/mirnavazquez/gut_microbiota/tree/main/R).

# You can find a complete description of the approach [here](https://mirnavazquez.netlify.app/media/gut_microbiota#1).